popovers: Redesign stream and topic action popover.

sayamsamal opened the pull request:
<!-- Describe your pull request here.-->

Redesigns the stream and topic action popover menu using the new **popover-menu** tippy theme. This is a part of the larger effort to redesign all the popover menus along the lines of the new gear menu/personal menu/help menu design. Also includes the new icons to go along with the redesign.

The redesign of topic visibility tab picker located in the topic action popover is being worked in #29479.

Fixes part of https://github.com/zulip/zulip/issues/28699.

[CZO discussion](https://chat.zulip.org/#narrow/stream/101-design/topic/UI.20redesign.3A.20stream.2Ftopic.20actions.20menu.20popover.20.2329785/near/1783805)

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

### Stream Actions Popover

- Light Mode

| Before | After |
|--------|--------|
| ![image](https://github.com/zulip/zulip/assets/82862779/b67d718a-9aff-4141-90f3-367474768543) | ![image](https://github.com/zulip/zulip/assets/82862779/266bc2cb-9616-481d-8ff1-7e8883c732fe) |

- Dark Mode

| Before | After |
|--------|--------|
| ![image](https://github.com/zulip/zulip/assets/82862779/9d845e8a-423c-43f0-ac19-603c2489dbe4) | ![stream_menu-dark-after](https://github.com/zulip/zulip/assets/82862779/bc05e8a2-92fb-40cd-8616-d0610503aad5) |

- With Color Picker

| Before | After |
|--------|--------|
| ![image](https://github.com/zulip/zulip/assets/82862779/21cdf6c2-68ea-46a9-8ca4-54a65c8193d5) | ![stream_menu-color_picker-after](https://github.com/zulip/zulip/assets/82862779/c9167ead-4c79-4a6f-87e2-565012ee1b09) |

- Long Stream Name (Wrapped)

| Before | After |
|--------|--------|
| ![stream_menu-wrapping-before](https://github.com/zulip/zulip/assets/82862779/a8f073da-9826-46c8-8f16-bddaf2d36f1c) | ![stream_menu-wrapping-after](https://github.com/zulip/zulip/assets/82862779/5549b734-1118-4683-94ea-c4efe51ce3c6) |

### Topic Actions Popover

- Light Mode

| Before | After |
|--------|--------|
| ![image](https://github.com/zulip/zulip/assets/82862779/55a99629-370d-4dcf-a9e1-cd5412bfebcc) | ![topic_menu-light-after](https://github.com/zulip/zulip/assets/82862779/54b2a246-9510-4b86-b468-0546a41e7009) |

- Dark Mode

| Before | After |
|--------|--------|
| ![image](https://github.com/zulip/zulip/assets/82862779/960e3ebb-70ca-4d09-a3fe-734ca92d63cf) | ![topic_menu-dark-after](https://github.com/zulip/zulip/assets/82862779/7de5c4b8-1098-4711-a76e-b1912943b044) |

- Long Topic Name (Wrapped)

| Before | After |
|--------|--------|
| ![image](https://github.com/zulip/zulip/assets/82862779/a8cd0ac4-51d9-4af6-afac-781b996ebee2)| ![topic_menu-wrapping-after](https://github.com/zulip/zulip/assets/82862779/8153a165-1944-42ab-aacd-ae3f1f3cef17) |

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
I haven't tested since it's a draft, but the screenshots look nice! I guess some day we should rework the color picker, but it's not a big priority (#14961, and whatever design changes we make as part of that).

timabbott commented:
There's some vertical alignment issues, at least with "Unsubscribe" and its icon.

timabbott commented on web/src/stream_popover.js:
```diff
@@ -177,7 +178,7 @@ function build_stream_popover(opts) {
             // Choose a different color.
             $popper.on("click", ".choose_stream_color", (e) => {
-                const $popover = $(e.target).closest(".streams_popover");
+                const $popover = $(e.target).closest("#stream-actions-menu-popover");
```
Could this be using `instance` to avoid needing to do this kind of lookup?

sayamsamal replied:
Makes sense to avoid lookups via IDs, when we already have the `instance` object. While redesigning the popovers, I have been using IDs on the popovers since I was working under the assumption that we can have only one instance of a popover at a given time. If we want to keep the flexibility as you have suggested, I think it would make sense for me to use a unique classname instead of an ID on the popover. (Since otherwise, having multiple instances with the same ID would lead to invalid HTML).

sayamsamal replied:
I have updated the code to now use `$(instance.popper)` instead. Would you want me to put up a small commit which does the above ID -> CLASS conversion?

timabbott replied:
I think it's probably a worthwhile follow-up refactor but very low importance and not worth entangling with this PR, so leave it on your TODO list for potentially a follow-up pass.

sayamsamal commented:
@alya I agree on the need of a rework on the colorpicker. I noticed some issues with the current implementation,
1. While tabbing through the menu, we come across the cancel option, which has no text inside it. It basically resets the color selection to the previous default, in case we want to check what the previous color was before confirming. Due to such a small target area and no content inside, it is possible to trigger this only via tabbing through the menu or just spamming near that location with a cursor.
![image](https://github.com/zulip/zulip/assets/82862779/4ea7ecb4-d298-412d-aa43-d25a3969d5d0)
2. When we confirm a color, we don't close the color picker. Maybe that's a intentional design choice, since it might be irritating for a user to navigate back to the color picker after every change. But this might be worth a look.
3. The confirm button itself has an outdated/mismatched look when focused/clicked.
![Confirm-button-colorpicker](https://github.com/zulip/zulip/assets/82862779/0a4b37bf-cf8f-429d-a802-c92bfa8be9c2)

sayamsamal commented:
> There's some vertical alignment issues, at least with "Unsubscribe" and its icon.
> 
> I'm also not sure about the increased minimum width for this one.

@timabbott The icon misalignment is due to most of the icons being the older font-awesome icons. It will get auto fixed once I request the new icons from Vlad, thanks! I'll start a design topic at CZO so that we can iron out the other issues.

sayamsamal commented:
UPDATED the PR to also include the topic action popover redesign. (changes are almost similar to those of the stream actions redesign).

sayamsamal commented:
**Updates:**
- Added the new icons provided by Vlad, for both, the topic and stream action popovers.
- Restored `no-auto-hide-left-sidebar-overlay` class, which I had removed my mistake.
- Updated the PR to show the current state.

timabbott commented:
Merged, thanks @sayamsamal! Please do a before/after test of popovers popover in the spectators mode as an additional QA step -- I forgot to do that before merging, and it's a thing I could imagine breaking.

One immediate follow-up worth doing is renaming the files; we have `web/templates/popovers/*_popover.hbs` as the naming convention for modernized popovers, and these should get moved to that standard.
